version: '3.8'

services:
  db:
    image: postgres:15
    container_name: odoo18-db
    environment:
      - POSTGRES_USER=odoo
      - POSTGRES_PASSWORD=odoo
      - POSTGRES_DB=postgres
    volumes:
      - db_data:/var/lib/postgresql/data
    restart: unless-stopped
    ports:
      - "5432:5432"

  web:
    image: odoo:18.0
    container_name: odoo18-web
    depends_on:
      - db
    ports:
      - "8069:8069"
    environment:
      - HOST=db
      - USER=odoo
      - PASSWORD=odoo
    volumes:
      - odoo_data:/var/lib/odoo
      - ./config:/etc/odoo
      - ./enterprise:/mnt/enterprise-addons
      - ./odoo/addons:/mnt/extra-addons
      - ./odoo/outputs:/var/log/odoo
    restart: unless-stopped

  nginx:
    image: nginx:latest
    container_name: odoo-nginx
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx/conf/odoo-https.conf:/etc/nginx/conf.d/default.conf
      - ./nginx/inc:/etc/nginx/inc
      - ./nginx/ssl:/etc/nginx/ssl
      - ./nginx/certbot/www:/var/www/certbot
      - ./nginx/certbot/conf:/etc/letsencrypt
    depends_on:
      - web

  certbot:
    image: certbot/certbot
    container_name: odoo-certbot
    volumes:
      - ./nginx/certbot/www:/var/www/certbot
      - ./nginx/certbot/conf:/etc/letsencrypt

volumes:
  db_data:
  odoo_data:
